src/Surfnet/StepupRa/RaBundle/Controller/Vetting/SmsController.php

```diff
@@ -112,6 +113,12 @@ public function provePossessionAction(Request $request, $procedureId)
             ->createForm('ra_verify_phone_number', $command, ['procedureId' => $procedureId])
             ->handleRequest($request);
 
+        /** @var SubmitButton $cancelButton */
+        $cancelButton = $form->get('cancel');
+        if ($cancelButton->isClicked()) {
+            return $this->forward('SurfnetStepupRaRaBundle:Vetting:cancelProcedure', ['procedureId' => $procedureId]);
```

Comment:
Hmm, this `forward` depends on the fact that both actions require a POST, and that the cancelProcedureAction has a procedureId parameter. Yet, it's not very clear, from the point of view of the cancelProcedure action, that it is depended upon in that way. In other words: we would probably encounter errors during run-time should we modify the cancelProcedure in the future.

What about cancelling the procedure from this action? At least the application won't break run-time.

```php
if ($cancelButton->isClicked()) {
    $this->getVettingService()->cancelProcedure($procedureId);
    $this->addFlash('info', $this->get('translator')->trans('ra.vetting.flash.cancelled'));

    return $this->redirectToRoute('ra_vetting_search');
}
```